kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/model/ingress/IngressModel.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator.model.ingress;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import io.fabric8.kubernetes.api.model.ContainerPort;
+import io.fabric8.kubernetes.api.model.ServiceBuilder;
+
+import io.kroxylicious.kubernetes.api.v1alpha1.virtualkafkaclusterspec.ingresses.Tls;
+import io.kroxylicious.proxy.config.NodeIdentificationStrategy;
+
+/**
+ * Kubernetes Ingress model representing a single VirtualCluster Gateway in the Proxy Config
```

Comment:
"Kubernetes Ingress model" is really confusing, because this is not referring to the Kubernetes `Ingress` resource. I would just drop that bit entirely

```suggestion
 * Represents a single VirtualCluster Gateway in the Proxy Config
```